Core/HistoManager.cpp

```diff
@@ -27,29 +27,90 @@ HistoManager::HistoManager() : fMakeControlHisto(true)
 
 HistoManager::~HistoManager() {}
 
+void HistoManager::createHistogramWithAxes(TObject* object, TString xAxisName, TString yAxisName, TString zAxisName) 
+{ 
+  TClass *cl = object->IsA();
+  if (cl->InheritsFrom("TH1D"))
+  {
+    TH1D* tempHisto = dynamic_cast<TH1D*>(object);
+    tempHisto->GetXaxis()->SetTitle(xAxisName);
+    tempHisto->GetYaxis()->SetTitle(yAxisName);
+  }
+  else if (cl->InheritsFrom("TH2D"))
+  {
+    TH2D* tempHisto = dynamic_cast<TH2D*>(object);
+    tempHisto->GetXaxis()->SetTitle(xAxisName);
+    tempHisto->GetYaxis()->SetTitle(yAxisName);
+  }
+  else if (cl->InheritsFrom("TH3D"))
+  {
+    TH3D* tempHisto = dynamic_cast<TH3D*>(object);
+    tempHisto->GetXaxis()->SetTitle(xAxisName);
+    tempHisto->GetYaxis()->SetTitle(yAxisName);
+    tempHisto->GetZaxis()->SetTitle(zAxisName);
+  }
+  fStats.Add(object);
+}
+
+void HistoManager::fillHistogram(const char* name, double xValue, doubleCheck yValue, doubleCheck zValue)
+{
+  TObject *tempObject = getObject<TObject>(name);
+  if (!tempObject)
+  {
+    writeError(name, " does not exist");
+    return;
+  }
+  TClass *cl = tempObject->IsA();
+  if (cl->InheritsFrom("TH1D"))
+  {
+    TH1D* tempHisto = dynamic_cast<TH1D*>(tempObject);
+    tempHisto->Fill(xValue);
+  }
+  else if (cl->InheritsFrom("TH2D"))
+  {
+    TH2D* tempHisto = dynamic_cast<TH2D*>(tempObject);
+    if (yValue.isChanged)
+      tempHisto->Fill(xValue, yValue.value);
+    else
+      writeError(name, " does not received argument for Y axis" );
+  }
+  else if (cl->InheritsFrom("TH3D"))
+  {
+    TH3D* tempHisto = dynamic_cast<TH3D*>(tempObject);
+    if (zValue.isChanged)
+      tempHisto->Fill(xValue, yValue.value, zValue.value);
+    else if (yValue.isChanged)
+      writeError(name, " does not received argument for Y and Z axis");
+    else
+      writeError(name, " does not received argument for Z axis");
+  }  
```

Comment:
probably you should add last else and return error  that it does not inherit from any of the types you checked e.g,. if the object is TGraph or even TH1F it will not be an error currently